workbench-core/accounts/src/utilities/hostingAccountLifecycleService.test.ts

```diff
@@ -600,4 +626,113 @@ describe('HostingAccountLifecycleService', () => {
+      JSON.parse(`
+      {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Sid": "Deny requests that do not use SigV4",
+                "Effect": "Deny",
+                "Principal": {
+                    "AWS": "*"
+                },
+                "Action": "s3:*",
+                "Resource": "${sampleBucketArn}/*",
+                "Condition": {
+                    "StringNotEquals": {
+                        "s3:signatureversion": "AWS4-HMAC-SHA256"
+                    }
+                }
+            },
+            {
+                "Sid": "List:environment-files",
+                "Effect": "Allow",
+                "Principal": {
+                    "AWS": "arn:aws:iam::${sampleAccountId}:root"
+                },
+                "Action": "s3:ListBucket",
+                "Resource": "${sampleBucketArn}",
+                "Condition": {
+                    "StringLike": {
+                        "s3:prefix": "environment-files*"
+                    }
+                }
+            },
+            {
+                "Sid": "Get:environment-files",
+                "Effect": "Allow",
+                "Principal": {
+                    "AWS": "arn:aws:iam::${sampleAccountId}:root"
+                },
+                "Action": "s3:GetObject",
+                "Resource": "${sampleBucketArn}/environment-files*"
+            }
+        ]
+    }`)
+    );
+
+    const postUpdatedPolicy = hostingAccountLifecycleService['_updateBucketPolicyDocumentWithAllStatements'](
+      sampleBucketArn,
+      sampleAccountId,
+      basePolicy
+    );
+    expect(postUpdatedPolicy).toEqual(expectedPolicy);
+  });
+
+  test('buildTemplateURLForAccount basic unit test', async () => {
+    const service = new HostingAccountLifecycleService();
+    const sampleExternalId = 'sample';
+    const region = process.env.AWS_REGION!;
+
+    service['_aws'].clients.s3 = new S3({
+      region : region,
+      credentials: {
+        accessKeyId: "AKIAIOSFODNN7EXAMPLE",
+        secretAccessKey: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+      }
+    });
+
+    const testUrl = 'https://testurl.com'
+    const expectedCreateUrl = 'https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/create/review/?templateURL=https%3A%2F%2Ftesturl.com&stackName=swb-swbv2-va-hosting-account&param_Namespace=swb-swbv2-va&param_MainAccountId=123456789012&param_ExternalId=sample&param_AccountHandlerRoleArn=arn:aws:iam::123456789012:role/swb-swbv2-va-accountHandlerLambdaServiceRole-XXXXXXXXXXE88&param_ApiHandlerRoleArn=arn:aws:iam::123456789012:role/swb-swbv2-va-apiLambdaServiceRoleXXXXXXXX-XXXXXXXX&param_StatusHandlerRoleArn=arn:aws:events:us-east-1:123456789012:event-bus/swb-swbv2-va&param_EnableFlowLogs=true'
+
+    const cfnMock = mockClient(CloudFormationClient);
+    mockCloudformationOutputs(cfnMock);
+    (getSignedUrl as jest.Mock).mockImplementation(() => testUrl)
```

Comment:
It looks like we're just testing that the method is called. We should atleast test that the parameter being passed into the method is what we expect. In this case, it's important to test the `bucketName` and `key` is correct.